### PR TITLE
Support for duckdns

### PR DESCRIPTION
**Pull request by ludeeus**
Initial support for duckdns

### CODE REVIEW

File: package/opt/hassbian/suites/install_duckdns.sh [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+echo "Creating a script file to be used by cron"
+echo "echo url='https://www.duckdns.org/update?domains=$domain&token=$token&ip=' | curl -k -o ~/duckdns/duck.log -K -" > duck.sh
+
+echo "Setting premissions..."
+chmod 700 duck.sh
+
+echo "Creating cron job..."
+(crontab -l ; echo "*/5 * * * * ~/duckdns/duck.sh >/dev/null 2>&1")| crontab -
+
+echo "Moving back to root..."
```
Comment (Landrash) [inline package/opt/hassbian/suites/install_duckdns.sh]: Changing?

File: package/opt/hassbian/suites/install_duckdns.sh [hunk 2]
```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+function duckdns-show-short-info {
+echo -e "Setup for duckdns auto renewal"
+}
+
+function duckdns-show-long-info {
+echo -e "This script adds an cron job to uppdate you duckdns IP address."
```
Comment (Landrash) [inline package/opt/hassbian/suites/install_duckdns.sh]: It doesn't update the IP address ;)

File: package/opt/hassbian/suites/install_duckdns.sh [hunk 3]
```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+function duckdns-show-short-info {
+echo -e "Setup for duckdns auto renewal"
```
Comment (Landrash) [inline package/opt/hassbian/suites/install_duckdns.sh]: Duck DNS uses the following in their written material and we should try to follow it "Duck DNS".

File: package/opt/hassbian/suites/install_duckdns.sh [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+chmod 700 duck.sh
+
+echo "Creating cron job..."
+(crontab -l ; echo "*/5 * * * * ~/duckdns/duck.sh >/dev/null 2>&1")| crontab -
+
+echo "Moving back to root..."
+EOF
+
+echo "Starting cron service..."
+sudo service cron start
```
Comment (Landrash) [inline package/opt/hassbian/suites/install_duckdns.sh]: Cron is already running and should be restarted.
Try to use the systemd restart instead also to be consistent over our scripts.
`sudo systemctl restart cron.service`

File: package/opt/hassbian/suites/install_duckdns.sh [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+fi
+
+echo -n "Token: "
+read token
+echo
+if [ ! "$token" ]; then
+  exit
+fi
+
+echo "Changing back to pi user..."
```
Comment (Landrash) [inline package/opt/hassbian/suites/install_duckdns.sh]: It would be preferable if this was run by the `homeassistant` user.

**Review by Landrash — CHANGES_REQUESTED** [review]
Some minor reworking and rewording is preferable.

File: package/opt/hassbian/suites/install_duckdns.sh (hunk 3 above)
Comment (ludeeus) [reply package/opt/hassbian/suites/install_duckdns.sh]: Done :)

File: package/opt/hassbian/suites/install_duckdns.sh (hunk 2 above)
Comment (ludeeus) [reply package/opt/hassbian/suites/install_duckdns.sh]: Not sure how else to put it :(

Updated description to this:
This script adds an cron job to auto uppdate you the WAN IP address for the defined domain.

File: package/opt/hassbian/suites/install_duckdns.sh (hunk 5 above)
Comment (ludeeus) [reply package/opt/hassbian/suites/install_duckdns.sh]: Done :)

File: package/opt/hassbian/suites/install_duckdns.sh (hunk 1 above)
Comment (ludeeus) [reply package/opt/hassbian/suites/install_duckdns.sh]: That is better, done ;)

File: package/opt/hassbian/suites/install_duckdns.sh (hunk 4 above)
Comment (ludeeus) [reply package/opt/hassbian/suites/install_duckdns.sh]: Done ;)

File: package/opt/hassbian/suites/install_duckdns.sh [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+echo
+if [ ! "$token" ]; then
+  exit
+fi
+
+echo "Changing back to homeassistant user..."
+sudo -u homeassistant -H /bin/bash << EOF
+cd
+
+echo "Creating duckdns foler..."
```
Comment (Landrash) [inline package/opt/hassbian/suites/install_duckdns.sh]: *folder

File: package/opt/hassbian/suites/install_duckdns.sh [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+echo "Setting premissions..."
+chmod 700 duck.sh
+
+echo "Creating cron job..."
+(crontab -l ; echo "*/5 * * * * ~/duckdns/duck.sh >/dev/null 2>&1")| crontab -
+
+echo "Changing back to root user..."
+EOF
+
+echo "Resarting cron service..."
```
Comment (Landrash) [inline package/opt/hassbian/suites/install_duckdns.sh]: *Restarting

File: package/opt/hassbian/suites/install_duckdns.sh [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+function duckdns-show-short-info {
+echo -e "Setup for Duck DNS auto renewal"
+}
+
+function duckdns-show-long-info {
+echo -e "This script adds an cron job to auto uppdate you the WAN IP address for the defined domain."
+}
+
+function duckdns-show-copyright-info {
+	echo "Original consept by Ludeeus <https://github.com/ludeeus>"
```
Comment (Landrash) [inline package/opt/hassbian/suites/install_duckdns.sh]: *Concept

Comment (ludeeus) [reply package/opt/hassbian/suites/install_duckdns.sh]: Done :)

File: package/opt/hassbian/suites/install_duckdns.sh (hunk 6 above)
Comment (ludeeus) [reply package/opt/hassbian/suites/install_duckdns.sh]: Done :)

File: package/opt/hassbian/suites/install_duckdns.sh (hunk 7 above)
Comment (ludeeus) [reply package/opt/hassbian/suites/install_duckdns.sh]: Done :)

**Review by Landrash — APPROVED** [review]
Looks good!

**Pull request merged**